playbooks/common/openshift-node/config.yml

```diff
@@ -84,6 +87,7 @@
     args:
       creates: "{{ etcd_generated_certs_dir }}/{{ item.etcd_cert_subdir }}.tgz"
     with_items: etcd_needing_client_certs
+    when: etcd_client_flannel_certs_missing is defined and etcd_client_flannel_certs_missing
```

Comment:
Have you tested this? I would expect it to throw an error, since with_items should be evaluated before the when (which is evaluated on each loop iteration)